lib/server.js

```diff
@@ -48,6 +49,21 @@ function createServer (options, httpHandler) {
     } else {
       listenOptions.cb = cb
     }
+    if (listenOptions.signal) {
+      if (typeof listenOptions.signal.on !== 'function' && typeof listenOptions.signal.addEventListener !== 'function') {
+        throw new FST_ERR_LISTEN_OPTIONS_INVALID('Invalid options.signal')
+      }
+
+      if (listenOptions.signal.aborted) {
+        this.close()
+      } else {
+        const onAborted = () => {
+          this.close()
+        }
+        listenOptions.signal.addEventListener('abort', onAborted, { once: true })
+        server.once('close', () => listenOptions.signal.removeEventListener('abort', onAborted))
+      }
```

Comment:
Sorry for the constant comments, but there are some gotchas that I figured out recently.

When calling `abortcontroller.abort` if passing a signal, will cause a similar effect on the server as calling `server.close`.

From Node documentation:
> If the signal option is enabled, calling .abort() on the corresponding AbortController is similar to calling .close() on the server:


For instance, we need to be careful and mimic that behavior to also execute the `close` process of `fastify`.
From your changes, you've them pretty much covered. The only thing we should care about next is to remove the `onAborted` listener as soon as the event is triggered and we call `fastify.close` and not after the serve has been `closed`.

wdyt?

Reply:
@metcoder95 how do you think we should handle if signal is already aborted? should we also remove it before triggering the close?

Reply:
@metcoder95 I just realized I don't really need to remove it by myself since i'm using the once options:

> once Optional

    A boolean value indicating that the listener should be invoked at most once after being added. If true, the listener would be automatically removed when invoked. If not specified, defaults to false.